src/arm/debugger/debugger.c

```diff
@@ -69,10 +107,42 @@ static bool ARMDebuggerUpdateStackTraceInternal(struct mDebuggerPlatform* d, uin
 		}
 		destAddress = info.op1.immediate + cpu->gprs[ARM_PC];
 	} else if (info.operandFormat & ARM_OPERAND_REGISTER_1) {
-		if (!isCall && info.op1.reg != ARM_LR && !(_ARMModeHasSPSR(cpu->cpsr.priv) && info.op1.reg == ARM_PC)) {
-			return false;
+		if (isCall) {
+			destAddress = cpu->gprs[info.op1.reg];
+		} else {
+			bool isExceptionReturn = _ARMModeHasSPSR(cpu->cpsr.priv) && info.affectsCPSR && info.op1.reg == ARM_PC;
+			bool isMovPcLr = (info.operandFormat & ARM_OPERAND_REGISTER_2) && info.op1.reg == ARM_PC && info.op2.reg == ARM_LR;
```

Comment:
Though I doubt anything does `cmp pc, lr` (or `cmn`, `tst`, or `teq`), you should probably check `ARM_OPERAND_AFFECTED_1` as well.

Reply:
That's handled farther up the function by checking `info.branchType`.

Reply:
In that case you should probably fix the bugs that makes that get set when `ARM_OPERAND_AFFECTED_1` isn't: https://github.com/mgba-emu/mgba/blob/master/src/arm/decoder-arm.c#L112 and https://github.com/mgba-emu/mgba/blob/master/src/arm/decoder-thumb.c#L125

(Also it looks like decoding `ldr pc, [...]` doesn't set that value: https://github.com/mgba-emu/mgba/blob/master/src/arm/decoder-arm.c#L177)